hermit-macro/src/system.rs

```diff
@@ -59,13 +59,27 @@ fn parse_sig(sig: &Signature) -> Result<ParsedSig> {
 }
 
 fn validate_attrs(attrs: &[Attribute]) -> Result<()> {
+	let mut no_mangle_found = false;
 	for attr in attrs {
-		if !attr.path().is_ident("cfg") && !attr.path().is_ident("doc") {
+		if !attr.path().is_ident("cfg")
+			&& !attr.path().is_ident("doc")
+			&& !attr.path().is_ident("no_mangle")
+		{
 			bail!(
 				attr,
-				"#[system] functions may only have `#[doc]` and `#[cfg]` attributes"
+				"#[system] functions may only have `#[doc]`, `#[no_mangle]` and `#[cfg]` attributes"
 			);
 		}
+		if attr.path().is_ident("no_mangle") {
+			no_mangle_found = true;
+		}
```

Comment:
You could write this as `no_mangle_found = attr.path().is_ident("no_mangle");`

Reply:
No, we can't, this is a loop over multiple attrs.

Reply:
Yes, we can. Obviously, this is part of the loop.

Reply:
ah, now I see. Nevermind.